core/save-git.c

```diff
@@ -596,6 +596,7 @@ static int save_one_divecomputer(git_repository *repo, struct dir *tree, struct
 	ret = blob_insert(repo, tree, &buf, "Divecomputer%c%03u", idx ? '-' : 0, idx);
 	if (ret)
 		report_error("divecomputer tree insert failed");
+	free_buffer(&buf);
```

Comment:
this is a double `free_buffer()`. look at `blob_insert()` - it already calls `free_buffer()`.
it's not ideal that the ownership is transferred to `blob_insert()`, too.